tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts

```diff
@@ -674,11 +716,55 @@ export class ScalarCardContainer implements CardRenderer, OnInit, OnDestroy {
     );
   }
 
-  editColumnHeaders(headerEditInfo: HeaderEditInfo) {
-    this.store.dispatch(dataTableColumnOrderChanged(headerEditInfo));
+  editColumnHeaders({
```

Comment:
The UI will not stop my from switching the position of an hparam and non hparam column but I believe this is not supported. I think we either need to stop the user from dragging columns beyond groups or else support mixing groups.

Imagine the following initial column order
```
[Run Name, Experiment Name, HP1, HP2]
```
Now the user drags the Run Name Column to the end
```
[Run Name, Experiment Name, HP1, HP2, Run Name]
     ^-----------------------------------^
```
The resulting state will be quite different from what they expect
```
[Experiment Name, Run Name, HP1, HP2]
```

This isn't really the best example since Run isn't movable, but it should highlight the issue.

Reply:
I agree this is ideal. Turns out to be pretty simple to implement! Just prevent dragEnter if headers are in different groups, and we conveniently already have a function for checking groups from #6732

Result (it's hard to see the mouse, but it's hovering over the columns on the right):
![group_boundary](https://github.com/tensorflow/tensorboard/assets/736199/d7a5c8ad-e497-4b63-8663-0153190428c5)